pkg/app/piped/trigger/trigger.go

```diff
@@ -395,19 +425,21 @@ func isTouchedByChangedFiles(appDir string, dependencyDirs []string, changedFile
 	// this application is considered as touched.
 	for _, cf := range changedFiles {
 		if ok := strings.HasPrefix(cf, appDir); ok {
-			return true
+			return true, nil
 		}
 	}
 
-	// If any files inside the app's dependencies was changed
+	// If any changed files matches the specified "changes"
 	// this application is consided as touched too.
-	for _, depDir := range dependencyDirs {
-		for _, cf := range changedFiles {
-			if ok := strings.HasPrefix(cf, depDir); ok {
-				return true
-			}
+	for _, change := range changes {
+		matcher, err := filematcher.NewPatternMatcher([]string{change})
+		if err != nil {
+			return false, err
+		}
+		if matcher.MatchesAny(changedFiles) {
+			return true, nil
```

Comment:
Sorry if I'm dumb but is it needed to perform for loop? We just give all patterns to it, right?
```go
		matcher, err := filematcher.NewPatternMatcher([]string{changes})
		if err != nil {
			return false, err
		}
		if matcher.MatchesAny(changedFiles) {
```

Reply:
We can give all patterns to `filematcher` and parse all of them before checking.
But by doing one by one we can have fast-return and reduce the number of unneeded regex parsing.
(In the future we can enable the regex cache too.)

Reply:
Ah, that makes sense! Okay, let's keep it as is.